### ISSUE TITLE
AttributeError: module 'pandas' has no attribute 'compat'

### ISSUE THREAD

**Issue by g7dhaliwal** [tensorflow/tensorflow#26266]
Hi,

I am trying to use ScipyOptimizerInterface() in the tensorflow, but it gave the following attribute error

AttributeError: module 'pandas' has no attribute 'compat'.

By going through the discussion threads at Tensorflow github page, I have upgarded "dask", downgraded pandas, reinstalled tensorflow and scipy packages. Unfortunately, it is still giving me same AttributeError. Does anyone having similar issues and can help me to resolve it? I can use tensorflow normally for other minimizations algorithms (tested ADAMS) but for scipy's BFGS implementation, I am getting this attribute error.

I am running code on Linux Centos system with python 3.6 and tensorflow 1.12.0. Version for pandas is 0.24.0. I tried to downgrade the pandas to 0.19.2 but it broke other parts of my code which use f2py library.

Any thoughts on how to fix this issue?

**Comment by hongyonggan** [comment]
I have just solved this problem. I had a similar issue, It may be a problem caused by package conflicts.  .You can try this in your console:

`conda upgrade --all -y`

This command above can solve the following two questions.

**Issue 1：**
`>>> import pandas
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/hyg/soft/anaconda3/lib/python3.6/site-packages/pandas/__init__.py", line 42, in <module>
    from pandas.core.api import *
  File "/Users/hyg/soft/anaconda3/lib/python3.6/site-packages/pandas/core/api.py", line 26, in <module>
    from pandas.core.groupby import Grouper
  File "/Users/hyg/soft/anaconda3/lib/python3.6/site-packages/pandas/core/groupby/__init__.py", line 1, in <module>
    from pandas.core.groupby.groupby import GroupBy  # noqa: F401
  File "/Users/hyg/soft/anaconda3/lib/python3.6/site-packages/pandas/core/groupby/groupby.py", line 20, in <module>
    import pandas.compat as compat
AttributeError: module 'pandas' has no attribute 'compat'`

**Issue 2：**
`In [3]: import pandas
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-38d4b0363d82> in <module>()
----> 1 import pandas

~/soft/anaconda3/lib/python3.6/site-packages/pandas/__init__.py in <module>()
     40 import pandas.core.config_init
     41 
---> 42 from pandas.core.api import *
     43 from pandas.core.sparse.api import *
     44 from pandas.tseries.api import *

~/soft/anaconda3/lib/python3.6/site-packages/pandas/core/api.py in <module>()
      8 from pandas.core.dtypes.missing import isna, isnull, notna, notnull
      9 from pandas.core.arrays import Categorical
---> 10 from pandas.core.groupby.groupby import Grouper
     11 from pandas.io.formats.format import set_eng_float_format
     12 from pandas.core.index import (Index, CategoricalIndex, Int64Index,

~/soft/anaconda3/lib/python3.6/site-packages/pandas/core/groupby/__init__.py in <module>()
      1 # flake8: noqa
----> 2 from pandas.core.groupby.groupby import (
      3     Grouper, GroupBy, SeriesGroupBy, DataFrameGroupBy
      4 )

~/soft/anaconda3/lib/python3.6/site-packages/pandas/core/groupby/groupby.py in <module>()
     47                                CategoricalIndex, _ensure_index)
     48 from pandas.core.arrays import ExtensionArray, Categorical
---> 49 from pandas.core.frame import DataFrame
     50 from pandas.core.generic import NDFrame, _shared_docs
     51 from pandas.core.internals import BlockManager, make_block

~/soft/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py in <module>()
     72                                    create_block_manager_from_arrays,
     73                                    create_block_manager_from_blocks)
---> 74 from pandas.core.series import Series
     75 from pandas.core.arrays import Categorical, ExtensionArray
     76 import pandas.core.algorithms as algorithms

~/soft/anaconda3/lib/python3.6/site-packages/pandas/core/series.py in <module>()
     65 from pandas.compat.numpy import function as nv
     66 
---> 67 import pandas.core.ops as ops
     68 import pandas.core.algorithms as algorithms
     69 

AttributeError: module 'pandas' has no attribute 'core'
`

**Comment by g7dhaliwal** [comment]
Thanks for your reply. I updated the conda as per your suggestion. I am having the following error now

AttributeError: type object 'scipy.interpolate.interpnd.array' has no attribute '__reduce_cython__'

**Comment by jvishnuvardhan** [comment]
Thank you for your post. We noticed you have not filled out the fields in the issue [template](https://github.com/tensorflow/tensorflow/issues/new?template=10-build-installation-issue.md). Could you update them if they are relevant in your case, or leave them as N/A? Along with the template, please provide as many details as possible to find the root cause of the issue. Are you getting the error during installation or during running some code. It would be great if you can provide a small code to reproduce the error. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by jvishnuvardhan** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will open a new issue. Thanks!

**Comment by art1** [comment]
For me downgrading pandas to 0.23.0 and using scipy 1.1.0 fixed this issue

**Comment by jvishnuvardhan** [comment]
@art1 thanks for providing a solution. Thanks!

**Comment by THEEJUNG** [comment]
@art1 Thank you. Your solution perfectly works for me.

**Comment by Z-Y00** [comment]
For 0.25,0.24,0.23, it seems you will never get pandas.compat

```
Warning

The pandas.core, pandas.compat, and pandas.util top-level modules are PRIVATE. Stable functionality in such modules is not guaranteed. 
```
as in 0.23 
https://pandas.pydata.org/pandas-docs/version/0.23/api.html?highlight=compat
 so. @art1 why downgrading works for you? I'm new to python

and in 0.24
https://pandas.pydata.org/pandas-docs/version/0.24/reference/index.html

and in stable(Now 0.25)
https://pandas.pydata.org/pandas-docs/stable/reference/index.html?highlight=compat

**Comment by j-kuechle** [comment]
In my case, this error was caused my mixing package installations via pip and conda. In my case, I had installed pandas via pip by mistake, while using conda as a main package manager.

To check use:
conda list 

If the Channel of a package is "pypi", it was installed via pip.
The solution was:
pip uninstall pandas
conda install pandas

**Comment by Vicky2603** [comment]
Thanks @hongyonggan , after upgrading conda it works fine.

**Comment by soumya997** [comment]
might sound silly, but putting `import pandas as pd` on the top of the import cell resolved my error, Im using pandas==1.1.5, python 3.7.0.
